modules/account-lib/src/coin/stx/transactionBuilderFactory.ts

```diff
@@ -0,0 +1,72 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BufferReader, deserializeTransaction, PayloadType } from '@stacks/transactions';
+import { BaseTransactionBuilderFactory } from '../baseCoin';
+import { ParseTransactionError, InvalidTransactionError } from '../baseCoin/errors';
+import { TransferBuilder } from './transferBuilder';
+import { WalletInitializationBuilder } from './walletInitializationBuilder';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+
+export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  from(raw: string): TransactionBuilder {
+    this.validateRawTransaction(raw);
+    try {
+      const tx = this.parseTransaction(raw);
+      switch (tx.stxTransaction.payload.payloadType) {
+        case PayloadType.TokenTransfer:
+          return this.getTransferBuilder(tx);
+        // TODO: add case of wallet
```

Comment:
Ah one big thing here is that we haven't handled any ContractCall types. We need to be able to handle interactions with the stacks staking contract. This will be fairly important to add